aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -34,12 +34,17 @@ static inline Tensor to_impl(const Tensor& self, const TensorOptions& options, b
 
   if (memory_format == MemoryFormat::Preserve) {
     if (self.is_non_overlapping_and_dense()) {
-      // Copy all strides
-      auto r = at::empty_strided(self.sizes(),
-                                 self.strides(),
-                                 options.memory_format(c10::nullopt).pinned_memory(pin_out));
-      r.copy_(self, non_blocking);
-      return r;
+       if (self.is_quantized()){
+         auto r = at::empty_quantized(self.sizes(), self, options);
+         r.copy_(self, non_blocking);
+         return r;
```

Comment:
can we move `return r` outside the if-else?

Reply:
it throws an error about r not being defined. I originally wanted to move both return and copy outside the if statement.

Reply:
you can first declare the Tensor before the branch I think:
```python
Tensor r;
if (...) {
    r = ...
} else {
   r = ...
}
...
```